packages/mui-system/src/cssVars/createCssVarsProvider.js

```diff
@@ -242,7 +242,21 @@ export default function createCssVarsProvider(options) {
         lightColorScheme,
         mode,
         setColorScheme,
-        setMode,
+        setMode:
+          process.env.NODE_ENV === 'production'
+            ? setMode
+            : (newMode) => {
+                if (theme.colorSchemeSelector === 'media') {
+                  console.error(
+                    [
+                      'MUI: The `setMode` function has no effect if `colorSchemeSelector` is `media` (`media` is the default value).',
```

Comment:
- () To match with how we reference functions in the docs?
- if -> when. If could create doubt about either this is true or not in the case the warning triggers?

```suggestion
                      'MUI: The `setMode()` function has no effect when `colorSchemeSelector` is `media` (`media` is the default value).',
```

Rule of error messages: https://www.notion.so/mui-org/Technical-writing-guidance-7e55b517ac2e489a9ddb6d0f6dd765de?pvs=4#85219822c3194b6f8a49ee08ea82b90a.